xbmc/addons/kodi-dev-kit/include/kodi/addon-instance/pvr/Providers.h

```diff
@@ -0,0 +1,190 @@
+  ///
+
+  /// @addtogroup cpp_kodi_addon_pvr_Defs_PVRProvider
+  ///@{
+
+  /// @brief **required**\n
+  /// Unique identifier for this provider.
+  void SetUniqueId(unsigned int uniqueId) { m_cStructure->iUniqueId = uniqueId; }
+
+  /// @brief To get with @ref SetUniqueId changed values.
+  unsigned int GetUniqueId() const { return m_cStructure->iUniqueId; }
+
+  /// @brief **required**\n
+  /// Name given to this provider.
+  void SetProviderName(const std::string& providerName)
+  {
+    strncpy(m_cStructure->strProviderName, providerName.c_str(),
+            sizeof(m_cStructure->strProviderName) - 1);
+  }
+
+  /// @brief To get with @ref SetProviderName changed values.
+  std::string GetProviderName() const { return m_cStructure->strProviderName; }
+
+  /// @brief **optional**\n
+  /// Provider type.
+  ///
+  /// Set to @ref PVR_PVR_PROVIDER_TYPE_UNKNOWN if the type cannot be
+  /// determined.
+  ///
+  /// --------------------------------------------------------------------------
+  ///
+  /// Example:
+  /// ~~~~~~~~~~~~~{.cpp}
+  /// kodi::addon::PVRRecording tag;
+  /// tag.SetProviderType(PVR_PROVIDER_TYPE_SATELLITE);
+  /// ~~~~~~~~~~~~~
+  ///
+  void SetProviderType(PVR_PROVIDER_TYPE providerType)
+  {
+    m_cStructure->providerType = providerType;
+  }
+
+  /// @brief To get with @ref SetProviderType changed values
+  PVR_PROVIDER_TYPE GetProviderType() const { return m_cStructure->providerType; }
+
+  /// @brief **optional**\n
+  /// Path to the provider icon (if present).
+  void SetIconPath(const std::string& iconPath)
+  {
+    strncpy(m_cStructure->strIconPath, iconPath.c_str(), sizeof(m_cStructure->strIconPath) - 1);
+  }
+
+  /// @brief To get with @ref SetIconPath changed values.
+  std::string GetIconPath() const { return m_cStructure->strIconPath; }
+  ///@}
+
+  /// @brief **optional**\n
+  /// The country codes for the provider.
+  ///
+  /// @note Use @ref PROVIDER_STRING_TOKEN_SEPARATOR to separate different countries.
+  /// ISO 3166 country codes required (e.g 'GB,IE,CA'). Empty string for undefined.
+  void SetCountry(const std::string& country)
+  {
+    strncpy(m_cStructure->strCountry, country.c_str(), sizeof(m_cStructure->strCountry) - 1);
+  }
+
+  /// @brief To get with @ref SetCountry changed values.
+  std::string GetCountry() const { return m_cStructure->strCountry; }
+  ///@}
+
+  /// @brief **optional**\n
+  /// The language codes for the provider.
+  ///
+  /// @note Use @ref PROVIDER_STRING_TOKEN_SEPARATOR to separate different languages.
+  /// RFC 5646 standard codes required (e.g.: 'en_GB,fr_CA'). Empty string for undefined.
+  void SetLanguage(const std::string& language)
+  {
+    strncpy(m_cStructure->strLanguage, language.c_str(), sizeof(m_cStructure->strLanguage) - 1);
+  }
+
+  /// @brief To get with @ref SetLanguage changed values.
+  std::string GetLanguage() const { return m_cStructure->strLanguage; }
```

Comment:
Plural?!

BTW: Would it make sense to let the c++ wrapper do the tokenizing/de-tokenizing of contries/languages?

Reply:
Yes, hopefully we can access StringUtils from here to make it easier.

Reply:
Done